src/dotnet/commands/dotnet-list/dotnet-list-package/ListPackageReferencesCommandParser.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Linq;
+using Microsoft.DotNet.Cli.CommandLine;
+using LocalizableStrings = Microsoft.DotNet.Tools.List.PackageReferences.LocalizableStrings;
+
+namespace Microsoft.DotNet.Cli
+{
+    internal static class ListPackageReferencesCommandParser
+    {
+        public static Command ListPackageReferences() => Create.Command(
+                "package",
+                LocalizableStrings.AppFullName,
+                Accept.ZeroOrOneArgument(),
+                CommonOptions.HelpOption(),
+                Create.Option("--outdated",
+                              LocalizableStrings.CmdOutdatedDescription),
+                Create.Option("--framework",
+                              LocalizableStrings.CmdFrameworkDescription,
+                              Accept.OneOrMoreArguments()
+                                    .With(name: LocalizableStrings.CmdFramework)
+                                    .ForwardAsSingle(o => $"--framework {string.Join("%3B", o.Arguments)}")),
```

Comment:
I don't think we're actually using the forwarded arguments since `ListPackageReferencesCommand.TransformArgs` is manually formatting the arguments.  We can either specify most of these arguments as forwarded and use `OptionValuesToBeForwarded`, or continue to manually build the command line in `ListPackageReferencesCommand.TransformArgs`.  Between the two, I think I favor using forwarded args since these command line options are really pass-through to NuGet.  We'll still need to do the "missing outdated" checks, though.

Reply:
I don't know the history as to why it's done this way, but the CLI needs the arguments in order `list package`, while the NuGet command requiers the order `package list`. I don't see how to do that with the parsed options, so I think we can't use option forwarding.

Reply:
Actually, I do know the reason the order is different. "nuget.exe list" lists all the packages from a nuget feed, so "nuget.exe package list" was added to list packages and do outdated checking.

Reply:
We can still collect just the options as forwarded arguments and append them to whatever command line NuGet will accept to accomplish the task.

https://github.com/dotnet/cli/blob/95c6eff6daa1a69f29c42b2d405400ad44bdec91/src/dotnet/ArgumentForwardingExtensions.cs#L31

This simply does what the logic in `ListPackageReferencesCommand.TransformArgs` is currently doing, although in a declarative way.

For `--framework` though, it sounds like it should be `ForwardAsMany` to allow a multiple options forwarded to NuGet.

If we aren't going to use `OptionValuesToBeForwarded`, then we can remove the `ForwardAsXXX` calls entirely.

Reply:
If we did keep the forwarded arguments, I would expect `TransformArgs` to look like this:

```c#
private string[] TransformArgs()
{
    var args = new List<string>
    {
        "package",
        "list",
    };

    args.Add(GetProjectOrSolution());

    args.AddRange(_appliedCommand.OptionValuesToBeForwarded());
}
```